### PR TITLE
POST/PUT requests to ServiceNow

### PR DESCRIPTION
**Pull request by KieranWardle**

### CODE REVIEW

File: src/main/java/uk/gov/ons/fsdr/client/ServiceNowClient.java [hunk 1]
```diff
@@ -22,15 +24,15 @@
...
 
-  public ResponseEntity<ServiceNowResponse> postEmployee(Employee employee) {
+  private String path = "api/now/table/sys_user";
 
-    String serviceNowPostUrl = serviceNowBaseUrl + "api/now/table/sys_user";
+  public ResponseEntity<ServiceNowResponse> postEmployee(Employee employee) {
 
     EmployeeDto employeeDto = convertEmployeeToDto(employee);
-
+    System.out.println(serviceNowBaseUrl + path);
```
Comment (garethTurner) [inline src/main/java/uk/gov/ons/fsdr/client/ServiceNowClient.java]: remove println

File: src/main/java/uk/gov/ons/fsdr/dto/EmployeeDto.java [hunk 2]
```diff
@@ -5,25 +5,29 @@
...
+  private String location;
+  @JsonProperty("u_postcode") private String postcode;
+  @JsonProperty("u_personal_mobile") private String personalPhone;
   @JsonProperty("u_line_manager") private String lineManager;
   @JsonProperty("u_employment_status") private String employmentStatus;
   @JsonProperty("u_job_type") private String jobType;
   @JsonProperty("u_external_user") private boolean externalUser;
+  @JsonProperty("u_contract_start_date") private LocalDateTime contractStartDate;
+  @JsonProperty("u_contract_end_date") private LocalDateTime contractEndDate;
+  @JsonProperty("user_name") private String roleId;
```
Comment (garethTurner) [inline src/main/java/uk/gov/ons/fsdr/dto/EmployeeDto.java]: this should still be called userName

File: src/main/java/uk/gov/ons/fsdr/scheduled/Ingester.java [hunk 3]
```diff
@@ -71,13 +71,21 @@ private void processFile(String fileName) throws IOException {
...
+            else {
+                var serviceNowResponseResponse = serviceNowClient.postEmployee(employee);
+                if (serviceNowResponseResponse.getStatusCode() == HttpStatus.CREATED) {
+                    employee.setServiceNowUserId(serviceNowResponseResponse.getBody().getSysId());
+                }
+                ingesterService.createEmployee(employee);
+            }
         }
-        employeeRepository.flush();
+        //employeeRepository.flush();
```
Comment (garethTurner) [inline src/main/java/uk/gov/ons/fsdr/scheduled/Ingester.java]: is this meant to be commented out?

File: src/main/java/uk/gov/ons/fsdr/scheduled/Ingester.java [hunk 4]
```diff
@@ -71,13 +71,21 @@ private void processFile(String fileName) throws IOException {
...
+                employee.setServiceNowUserId(oldEmployee.getServiceNowUserId());
+                ingesterService.updateEmployee(employee);
+                serviceNowClient.putEmployee(employee);
+            }
+            else {
+                var serviceNowResponseResponse = serviceNowClient.postEmployee(employee);
+                if (serviceNowResponseResponse.getStatusCode() == HttpStatus.CREATED) {
+                    employee.setServiceNowUserId(serviceNowResponseResponse.getBody().getSysId());
+                }
+                ingesterService.createEmployee(employee);
```
Comment (garethTurner) [inline src/main/java/uk/gov/ons/fsdr/scheduled/Ingester.java]: is this just calling `employeeRepository.saveAndFlush(employee);`?

File: src/main/java/uk/gov/ons/fsdr/service/impl/IngesterServiceImpl.java [hunk 5]
```diff
@@ -25,10 +25,8 @@
   @Transactional
   public void updateEmployee(Employee employee) {
```
Comment (spdiaz1000) [inline src/main/java/uk/gov/ons/fsdr/service/impl/IngesterServiceImpl.java]: create a EmployeeHistoryFactory class, move it to the factory package

**Pull request merged**